ndcube/wcs/wrappers/compound_wcs.py

```diff
@@ -0,0 +1,213 @@
+        input pixel dimensions to the underlying WCSes. This should have length
+        equal to the total number of pixel dimensions in all input WCSes and
+        have a maximum of the number of input pixel dimensions to the resulting
+        compound WCS -1 (counts from 0). For example ``(0, 1, 2, 1)`` would end
+        up with the second and fourth pixel dimensions in the input WCSes being
+        shared, so the compound WCS would have 3 pixel dimensions ``(2 + 1)``.
+        See :ref:`compound-model-mappings` for more examples of this input
+        format.
+    pixel_atol : `float`
+        A tolerance used to check that the resulting pixel coordinates from
+        ``world_to_pixel`` are the same from all WCSes.
+    """
+    def __init__(self, *wcs, mapping=None, pixel_atol=1e-8):
+        self._wcs = wcs
+
+        if not mapping:
+            mapping = tuple(range(self._all_pixel_n_dim))
+
+        if not len(mapping) == self._all_pixel_n_dim:
+            raise ValueError(
+                "The length of the mapping must equal the total number of pixel dimensions in all input WCSes.")
+
+        self.mapping = Mapping(mapping)
+        self.atol = pixel_atol
+
+        # Validate the pixel bounds and shape are consistent
+        self.pixel_bounds
+        self.pixel_shape
+
+    @property
+    def _all_pixel_n_dim(self):
+        return sum([w.pixel_n_dim for w in self._wcs])
+
+    @property
+    def pixel_n_dim(self):
+        return self.mapping.n_inputs
+
+    @property
+    def world_n_dim(self):
+        return sum([w.world_n_dim for w in self._wcs])
+
+    @property
+    def world_axis_physical_types(self):
+        return tuplesum([w.world_axis_physical_types for w in self._wcs])
+
+    @property
+    def world_axis_units(self):
+        return tuplesum([w.world_axis_units for w in self._wcs])
+
+    def pixel_to_world_values(self, *pixel_arrays):
+        pixel_arrays = self.mapping(*pixel_arrays)
+        world_arrays = []
+        for w in self._wcs:
+            pixel_arrays_sub = pixel_arrays[:w.pixel_n_dim]
+            pixel_arrays = pixel_arrays[w.pixel_n_dim:]
+            world_arrays_sub = w.pixel_to_world_values(*pixel_arrays_sub)
+            if w.world_n_dim > 1:
+                world_arrays.extend(world_arrays_sub)
+            else:
+                world_arrays.append(world_arrays_sub)
+        return tuple(world_arrays)
+
+    def world_to_pixel_values(self, *world_arrays):
+        pixel_arrays = []
+        for w in self._wcs:
+            world_arrays_sub = world_arrays[:w.world_n_dim]
+            world_arrays = world_arrays[w.world_n_dim:]
+            pixel_arrays_sub = w.world_to_pixel_values(*world_arrays_sub)
+            if w.pixel_n_dim > 1:
+                pixel_arrays.extend(pixel_arrays_sub)
+            else:
+                pixel_arrays.append(pixel_arrays_sub)
+
+        pixel_arrays = tuple(pixel_arrays)
+        # for i, ix in enumerate(self.mapping.mapping):
```

Comment:
Can these commented out lines be removed?

Reply:
hmm I should probably get back to this.

Reply:
This issues is still open.

Reply:
Indeed.